### PR TITLE
Issue #415 (Implement Transplanting Report)

### PR DESCRIPTION
**Pull request by s017274**
__Pull Request Description__

This pull request closes Issue #415 by implementing a functional Transplanting Report on the BarnKit tab in FarmData2. Using the code from seedingReport.html and other FD2 references, the Transplanting Report implemented here displays transplanting logs from the FD2 sample database based on a requested date range. It allows the user to filter these logs by area and crop, and displays a summary table of the report at the bottom of the page. Finally, the user is able to edit and delete individual logs from the table and download the report in .csv format.

This work is currently in progress by Phillip Mollerus, Ananjan Rijal, and Benjamin Warren in Professor Farhan Siddiqui's COMP 290 class at Dickinson College.

---
__Licensing Certification__

FarmData2 is a [Free Cultural Work](https://freedomdefined.org/Definition) and all accepted contributions are licensed as described in the LICENSE.md file. This requires that the contributor holds the rights to do so. By submitting this pull request __I certify that I satisfy the terms of the [Developer Certificate of Origin](https://developercertificate.org/)__ for its contents.

### CODE REVIEW

**Comment by braughtg** [comment]
It looks like you are off to a good start here. Your PR seems to contain some changes that are not related to the Active Planting Report (e.g. FD2School stuff) that should not be included in the PR.  Those changes should ultimately be removed from the PR.

**Comment by johnmaccormick** [comment]
The authors of this PR state that it closes [Issue 415](https://github.com/DickinsonCollege/FarmData2/issues/415). This suggests that it may be ready for review and potential merging to the main branch. Perhaps a project maintainer can initiate that review? (I have not taken the time to test this code myself.)

Note: This PR is based on work done in Professor Farhan Siddiqui's COMP 290 class at Dickinson College, Spring 2022.

**Comment by s017274** [comment]
Having seen Professor MacCormick's note about this PR, I would like to quickly make a few clarifications. I worked on this PR in Professor Siddiqui's class - the code fulfills most of what Matt requested in a new Transplanting Report tab, but it is missing several key features. This [kanban board](https://github.com/mollerup23/FarmData2/projects/1) goes into more detail, but the back-end work for deleting and editing rows in the report table is incomplete (those functions currently only work on the front end, they do not interact with the database correctly) and the user is unable to download the table as a .csv file. 

Additionally, we were unable to resolve an issue that Professor Braught mentioned in his earlier comment, that some of the commits on this PR are unrelated to Issue #415 and should be removed. We attempted to do this by creating a new PR and cherry-picking commits, but that failed. 

TL;DR This exact PR should not be merged, since it contains commits with work on the FD2_School modules from COMP 290. However, the code contained herein successfully implements a Transplanting Report tab in the BarnKit with the exception of the full edit and delete row functions and the download .csv function. Please feel free to reach out to me (warrenbe@dickinson.edu), one of my group members, or Professor Siddiqui for further clarity. Thanks!

**Comment by braughtg** [comment]
Closed as the course associated with this issue has concluded.  It will be preserved in the project history as a basis for implementation of this feature in the future.
